spacesuit/src/value.rs

```diff
@@ -126,10 +126,26 @@ impl Into<Scalar> for SignedInteger {
 }
 
 impl Add for SignedInteger {
-    type Output = SignedInteger;
+    type Output = Option<SignedInteger>;
 
-    fn add(self, rhs: SignedInteger) -> SignedInteger {
-        SignedInteger(self.0 + rhs.0)
+    fn add(self, rhs: SignedInteger) -> Option<SignedInteger> {
+        let s = self.0 + rhs.0;
+        if s <= (u64::max as i128) && s >= (u64::min as i128) {
```

Comment:
`u64::min` is 0 and `i64::min` is not what we need. We need exactly `-(u64::MAX as i128)` for minimum (first cast `2^64-1` to i128, then negate it to get `-(2^64-1)`).